libcontainer/intelrdt/mbm.go

```diff
@@ -0,0 +1,108 @@
+// +build linux
+
+package intelrdt
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+var (
+	// The flag to indicate if Intel RDT/MBM is enabled
+	isMbmEnabled bool
+
+	enabledMonFeatures monFeatures
+)
+
+type monFeatures struct {
+	mbmTotalBytes bool
+	mbmLocalBytes bool
+	llcOccupancy  bool
+}
+
+// Check if Intel RDT/MBM is enabled
+func IsMbmEnabled() bool {
+	return isMbmEnabled
+}
+
+func getMonFeatures(intelRdtRoot string) (monFeatures, error) {
+	file, err := os.Open(filepath.Join(intelRdtRoot, "info", "L3_MON", "mon_features"))
+	if err != nil {
+		return monFeatures{}, err
+	}
+	return parseMonFeatures(file)
+}
+
+func parseMonFeatures(file io.Reader) (monFeatures, error) {
+	s := bufio.NewScanner(file)
+
+	monFeatures := monFeatures{}
+
+	for s.Scan() {
+		if err := s.Err(); err != nil {
```

Comment:
please double check if it is a good idea to check scan errors in the loop, to my knowledge if any error appears during scanning then the loop is broken, I used to see following form of scan loop
```
for scanner.Scan() {
    fmt.Println(scanner.Text())
}
if scanner.Err() != nil {
    fmt.Println(scanner.Err())
}
```